### PR TITLE
Fix collection upsert types

### PR DESCRIPTION
**Pull request by ferhatelmas**

### CODE REVIEW

File: src/collections.ts [hunk 1]
```diff
@@ -293,15 +295,15 @@ export class Collections<
    * @param {UpsertManyCollectionRequest} data - A single json object that contains information of many collections
    * @return {Promise<UpsertCollectionAPIResponse<CollectionType>>}
    */
-  upsertMany(data: UpsertManyCollectionRequest) {
+  upsertMany(items: UpsertManyCollectionRequest) {
```
Comment (ferhatelmas) [inline src/collections.ts:298]: not needed but good for clarification

File: src/collections.ts [hunk 2]
```diff
@@ -27,15 +32,12 @@ export type SelectCollectionAPIResponse<CollectionType extends UR = UR> = APIRes
 
 export type UpsertCollectionAPIResponse<CollectionType extends UR = UR> = APIResponse & {
   data: {
-    [key: string]: {
-      data: CollectionType;
```
Comment (ferhatelmas) [inline src/collections.ts:31]: this nesting is wrong, should be flat

File: src/collections.ts [hunk 3]
```diff
@@ -293,15 +295,15 @@ export class Collections<
    * @param {UpsertManyCollectionRequest} data - A single json object that contains information of many collections
```
Comment (mahboubii) [inline src/collections.ts:295]:
```suggestion
   * @param {UpsertManyCollectionRequest} items - A single json object that contains information of many collections
```

File: src/collections.ts [hunk 4]
```diff
@@ -7,6 +7,11 @@ type BaseCollection<CollectionType> = {
   id: string;
 };
 
+export type CollectionInput<CollectionType extends UR = UR> = CollectionType & {
```
Comment (mahboubii) [inline src/collections.ts:10]: CollectionInput is also used for API responses, maybe better to call it `CollectionItem`?

Comment (ferhatelmas) [reply src/collections.ts]: done

**Review by mahboubii — APPROVED** [review]

**Pull request merged**